VladMeleshko/nodejs-course-2021Q2
# Task 8: Authentification & JWT

1. Task: https://github.com/rolling-scopes-school/basic-nodejs-2021Q2/blob/master/descriptions/authentification-jwt.md
2. Done: 27.06.2021 / deadline: 27.06.2021
3. Self-test ( 100 / 100 ):
[x] Пароли пользователей сохраняются в базу в виде хэша с использованием bcrypt. ( 20 / 20 )
[x] Добавлен роут /login, связанная с ним логика разделена между контроллером (middleware) и соответствующим сервисом. В случае отсутствия юзера в БД, возвращается 403 (Forbidden) HTTP статус. ( 20 / 20 )
[x] JWT токен содержит userId и login, секретный ключ хранится в .env. ( 20 / 20 )
[x] Доступ ко всем роутам, за исключением /login, /doc и /, требует аутентификации. ( 20 / 20 )
[x] Проверка на наличие токена в реквесте реализована в отдельной middleware на уровне приложения. В случае если токен не валидный, или отсутствует, возвращается 401 (Unauthorized) HTTP статус. ( 20 / 20 )

- Штрафов нет.
- Общая оценка - 100.

User with login='admin' and password='admin' is create in migrations, when you run `npm run start-migration` locally or automatically in docker (check instruction below).

How to check project local:

- verifie environment variables on .env;
- setup empty base created on .env;
- run `npm run start-migration`.

How to check project in docker:

- run `docker-compose up --build`.

If you first ran the application in docker, and then want to run it locally, start the server with Nodemon using command `npm run start-nodemon:bcrypt` or without it using command `npm run start:bcrypt`.
